ui/src/tasks/components/TaskTokenDropdown.tsx

```diff
@@ -15,21 +15,43 @@ interface Props {
 
 export default class TaskTokenDropdown extends PureComponent<Props> {
   public render() {
-    const {tokens, selectedToken, onTokenChange} = this.props
+    const {onTokenChange} = this.props
 
     return (
       <Dropdown
-        selectedID={selectedToken.id}
+        selectedID={this.selectedID}
         buttonColor={ComponentColor.Primary}
         buttonSize={ComponentSize.Small}
         onChange={onTokenChange}
       >
-        {tokens.map(t => (
-          <Dropdown.Item id={t.id} key={t.id} value={t}>
-            {t.description || 'Name this token'}
-          </Dropdown.Item>
-        ))}
+        {this.dropdownTokens}
       </Dropdown>
     )
   }
+  private get dropdownTokens(): JSX.Element[] {
```

Comment:
I find that getters that return `JSX` are generally better as separate components.

Reply:
I think the dropdown complains if the items are not in an array and it complains if it's in a separate. Also, I think this would keep it consistent with other drop downs in the ui.